src/main/java/software/amazon/awssdk/crt/io/SocketOptions.java

```diff
@@ -97,7 +104,42 @@ int getValue() {
      * Creates a new set of socket options
      */
     public SocketOptions() {
+    }
+
+    /**
+     * Enables TCP keepalive.
+     *
+     * @param keepAliveTimeoutSecs Sets the number of seconds to wait for a keepalive response before considering the
+     * connection timed out. 0 disables keepalive.
+     * @param keepAliveIntervalSecs Sets the number of seconds between TCP keepalive packets being sent to the peer.
+     * 0 disables keepalive.
+     */
+    public void setTcpKeepAlive(int keepAliveTimeoutSecs, int keepAliveIntervalSecs)
+    {
+        if (keepAliveTimeoutSecs == 0 || keepAliveIntervalSecs == 0) {
+            Log.log(Log.LogLevel.Warn, Log.LogSubject.IoSocket,
+                    "Both keepAliveTimeoutSecs and keepAliveIntervalSecs must be non-zero in order to enable TCP keepalive");
+        }
+        this.keepAliveTimeoutSecs = keepAliveTimeoutSecs;
+        this.keepAliveIntervalSecs = keepAliveIntervalSecs;
+        this.keepAlive = true;
```

Comment:
Does this have to be set to false, if one of the numbers is zero or does it not matter?